src/app/App.jsx

```diff
@@ -0,0 +1,157 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withRouter, matchPath } from 'react-router-dom';
+
+import Base from 'terra-base';
+import Image from 'terra-image';
+import i18nSupportedLocales from 'terra-i18n/lib/i18nSupportedLocales';
+import ThemeProvider from 'terra-theme-provider';
+import NavigationLayout from 'terra-navigation-layout';
+
+import ApplicationHeader from './ApplicationHeader';
+import './App.scss';
+
+const propTypes = {
+  /**
+   * The title branding of the site.
+   */
+  appTitle: PropTypes.string,
+  /**
+   * The subtitle to be appended to the title.
+   */
+  appSubtitle: PropTypes.string,
+  /**
+   * The source of the logo element to be placed at the start of the toolbar.
+   */
+  appLogoSrc: PropTypes.string,
+  /**
+  * The configuration Object that will be used to generate the specified regions of the terra-navigation-layout.
+  * Note: The config prop is treated as an immutable object to prevent unnecessary processing and improve performance.
+  * If the configuration is changed after the first render, a new configuration object instance must be provided.
+  */
+  routeConfig: PropTypes.object.isRequired,
+  /**
+   * The navigaion links to display within the menu in the toolbar.
+   */
+  navigation: PropTypes.object.isRequired,
+  /**
+   * The root path for the site.
+   */
+  rootPath: PropTypes.string.isRequired,
+  /**
+   * The theme options the site should display in the theme utility in the toobar.
+   */
+  themes: PropTypes.object,
+  /**
+  * The locale options the site should display in the locale utility in the toobar.
+   // we should check these values are contained in i18nSupportedLocales
+   */
+  locales: PropTypes.array,
+  /**
+   * Whether or not to display the directionality utility in the toolbar.
+   */
+  hideBidiUtility: PropTypes.bool,
+  /**
+   * The directionality the site should default to. Either 'ltr' or 'rtl'.
+   */
+  defaultDir: PropTypes.string,
+  /**
+   * The theme the site should default to.
+   */
+  defaultTheme: PropTypes.string,
+  /**
+   * Injected by react-routed: represent where the app is now, where you want it to go,
+   * or even where it was.
+   */
+  location: PropTypes.shape({
+    pathname: PropTypes.string,
+  }),
+};
+
+const defaultProps = {
```

Comment:
These end up being listed twice. I should pull these in from the site.config file.